omc3/tbt/reader_trackone.py

```diff
@@ -69,12 +71,8 @@ def get_structure_from_trackone(nturns=0, npart=0, infile='trackone'):
             parts = l.split()
             if parts[0] == '#segment':
                 bpm_name = parts[-1].upper()
-                if ('BPM' in bpm_name) and (bpm_name not in bpms.keys()):
+                if (np.all([k not in bpm_name.lower() for k in ['start', 'end']])) and (bpm_name not in bpms.keys()):
                     bpms[bpm_name] = np.empty([npart, nturns, 8], dtype=float)
-            elif 'BPM' in bpm_name:
-                bpms[bpm_name][int(parts[0]) - 1, int(parts[1]) - 1, :] = np.array(parts[2:])
+            elif (np.all([k not in bpm_name.lower() for k in ['start', 'end']])):
+                bpms[bpm_name][np.abs(int(float(parts[0]))) - 1, int(float(parts[1])) - 1, :] = np.array(parts[2:])
```

Comment:
why the absolute value?

Reply:
in case you use the only_average option in madx, the particle id is saved as -1, in order to make indexing work the abs is introduced